app/src/main/java/dev/devdigi/music/connection/ServerConnection.kt

```
@@ -0,0 +1,116 @@
+package dev.devdigi.music.connection
+
+import java.net.URI
+
+data class ServerEndpoint(val value: String) {
+    companion object {
+        fun parse(
```

detekt (1.23.8)

[warning] The function parse appears to be too complex based on Cyclomatic Complexity (complexity: 27). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CyclomaticComplexMethod)